src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LocalPath.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.Helpers;
+
+/// <summary>
+/// Smart file path wrapper that allows easy combining and makes sure separators conform to the desired way.
+/// They support concatenation where the left-most (root) path decides the shape of the outcome.
+/// (e.g. "C:\foo" / "src/something" returns "C:\foo\src\something")
+/// 
+/// The concatenation works thanks to overloading of the / operator:
+///     var combinedPath = rootPath / "src" / "MyProject.csproj";
+/// </summary>
+public abstract class LocalPath
```

Comment:
LocalPath is not a bad name, but I'd propose to use `SmartPath` as on Linux, technically, you could have NFS mounts which are not local mounted at any point of the FS and you'd be able to handle these paths as well. Also, SmartPath (at least for me), has slight connotation with SmartPointers from C++ that also do "something extra" on top of regular pointers, similarly as this concept does for paths.

Reply:
Technically, I think you're right but looking at it from a more realistic point of view.. I think `LocalPath` is much more descriptive for what it represents as we don't really expect/care about network disks in our limited VMR scenario?